lib/factory_girl/linter_base.rb

```diff
@@ -0,0 +1,49 @@
+module FactoryGirl
+  class LinterBase
```

Comment:
instead of the template method pattern, could we instead use the strategy pattern, where the linter itself can be calculated where the flag of including traits (or not) passes in different strategies for building out the list of things to be linted? I think the code here is close to that already, but it removes the reliance on inheritance here, which I think we should shoot for.